### PR TITLE
Add update version action to po smart contract

### PR DESCRIPTION
**Pull request by shannynalayna**
This adds an implementation to the po smart contract for the "update version" action.

### CODE REVIEW

File: contracts/purchase_order/src/handler.rs [hunk 1]
```diff
@@ -627,12 +628,155 @@ fn create_version(
...
+            original_version.workflow_status(),
+            desired_state,
+        )));
+    }
+
+    let desired_workflow_state = version_subworkflow.state(desired_state).ok_or_else(|| {
+        ApplyError::InternalError("Unable to get state from subworkflow".to_string())
+    })?;
+
+    if desired_workflow_state.has_constraint(&WorkflowConstraint::Complete.to_string()) {
```
Comment (leebradley) [inline contracts/purchase_order/src/handler.rs:711]: I'm thinking most of this validation would be useful regardless of whether the version is complete.

**Review by leebradley — CHANGES_REQUESTED** [review]

File: contracts/purchase_order/src/handler.rs (hunk 1 above)
Comment (shannynalayna) [reply contracts/purchase_order/src/handler.rs:711]: I'm gonna put up a PR to add validation for all of the PO payloads before they're handed off into the individual funcs and then will rebase this one :) without this check probably

File: contracts/purchase_order/src/state.rs [hunk 2]
```diff
@@ -185,7 +185,27 @@ impl<'a> PurchaseOrderState<'a> {
...
         // Add the `PurchaseOrderVersion` to the purchase order
         let mut versions = orig_po.versions().to_vec();
-        versions.push(purchase_order_version);
+        let mut vers_index = None;
+        for (i, vers) in versions.iter().enumerate() {
+            if vers.version_id() == purchase_order_version.version_id() {
+                vers_index = Some(i);
+                break;
+            }
+        }
```
[the remark this answers is not in the document]
Comment (shannynalayna) [reply contracts/purchase_order/src/state.rs:194]: Thanks for this!

**Comment by shannynalayna** [comment]
This will need to be rebased on #1015

**Review by leebradley — APPROVED** [review]

**Review by leebradley — DISMISSED** [review]

**Review by dnewh — DISMISSED** [review]

**Review by leebradley — APPROVED** [review]

**Review by agunde406 — APPROVED** [review]

**Pull request merged**